### ISSUE TITLE
1.5b not extracting .dmg on 10.12

### ISSUE THREAD

**Issue by sanfords** [sparkle-project/Sparkle#897]
Anybody having issues with 1.5b not extracting dmg files on 10.12?  I have many installs of 1.5b in the wild.  Users are reporting `Sparkle Error. An error has arisen during the archive data extraction. Please try again later.`

I switched my appcast to download a .zip instead, and it's working again.

Something change in 10.12 around this?

**Comment by jkbullard** [comment]
Yes, I had the same problem with [Tunnelblick](https://tunnelblick.net) -- it was present in the first Sierra beta. Instead of providing .zips, I upgraded the application to a newer version of Sparkle, which doesn't have the problem.

**Comment by pornel** [comment]
Please don't use 1.5b. It has critical security vulnerability. It has known incompatibilities that will never be fixed.

**Comment by sanfords** [comment]
@pornel please tell me how to upgrade 1.5b for users who have had it in the wild for 5 years.

**Comment by pornel** [comment]
In Sierra there are system-level changes to mounted images (there's extra sandboxing), but it shouldn't affect Sparkle when the app is run normally.  

If ZIP works, then the problem can't be in signing and authorization, but we've changed the DMG unarchiver only recently in 115686cd07beceeeb923f22fbcc62af7876786b8, so I'm not sure what's going on there exactly.

**Comment by sanfords** [comment]
Might be interesting to revert than branch... out of curiosity.

**Comment by jkbullard** [comment]
@pornel, @sanfords - I believe the problem (not extracting properly on Sierra) was fixed before 115686c because I did testing with various recent releases of Sparkle during August and September. All the releases I tested fixed the problem (and I think I went back to 1.12 or so).

@sanfords - As I wrote above, I finally decided to start using a more recent Sparkle version for [Tunnelblick](https://tunnelblick.net). I ended up building from the 077c644 commit made 2016-09-03.

I released updates with the new Sparkle before Sierra was released (too late for you to do that now), and put up a notice on our home page and several other places explaining why it was advisable to update Tunnelblick before upgrading to Sparkle.

One option (which I decided against) would be to update your app to use the new Sparkle and a different appcast URL (which uses a .dmg). Then modify your original appcast to use a .zip to update to the latest version of your app (the one with the latest Sparkle and new appcast URL).

**Comment by sanfords** [comment]
A sparkle URL should include OS version and other goodies so a cgi can make an informed decision.

**Comment by zorgiepoo** [comment]
I briefly looked at the code from 1.5 betas and it looks like it used to try creating a directory inside `/Volume/` before mounting the disk image, but that's no longer allowed under Sierra, and I'm not sure how that ever could have worked under a non-admin user.

A simple solution is to use zip or another archive format, which I personally believe is better for serving updates anyway.

@sanfords you can customize the `SUUpdater`'s `httpHeaders` property for additional information. Note Sparkle also has an option for including system profile information but usually the user has to opt into it when confirming to check for updates, and that information is not sent for every request. There are potential privacy issues with just assuming additional information can be sent without consent as well.

**Comment by sanfords** [comment]
Someone should make sure the zip option works under future OS betas.  And if not, contact Apple with a bug report.  Maybe Apple can work around it.  Sparkle is very popular, and I suspect 1.5b is very popular in the field.  If 1.5b were broken by an OS update, a lot of users would be affected.

**Comment by sanfords** [comment]
@zorgiepoo Thanks for that investigation.  I was curious.  My own appcast URL is has enough tidbits so my cgi can make an informed decision.

**Comment by zorgiepoo** [comment]
Apple's new security mechanism is not going to work around a bug we had because of a poor assumption we made in a really old version of Sparkle. The defect is also not fatal and can be simply worked around.

We have automated unit tests for unarchiving code, but one can't expect volunteers to do all sorts of manual labor especially for ancient versions. App developers also need to do their part by testing and reporting bugs under beta OS's if that's important to them, and by also being informed of updates to their dependencies ;).

1.5b may also not be as popular as you suspect (to apps still being maintained) because of the aforementioned security vulnerability that affected many apps and generated lot of attention.
